openebs/pkg/volume/maya_api.go

```diff
@@ -109,24 +109,20 @@ func (v OpenEBSVolume) CreateVolume(vs mayav1.VolumeSpec) (string, error) {
 	resp, err := c.Do(req)
 	if err != nil {
 		glog.Errorf("Error when connecting maya-apiserver %v", err)
-		return "Could not connect to maya-apiserver", err
+		return err
 	}
 	defer resp.Body.Close()
-
+	code := resp.StatusCode
+	if code != http.StatusOK {
+		return errors.New(http.StatusText(code))
```

Comment:
Can we get the original error message sent from api server. I guess we are not retrieving it.